### ISSUE TITLE
python approach for 1.4.3.6

### ISSUE THREAD

**Issue by franktreppromo65** [TruePikachu/terraria-map-dump#10]
hi, i am currently doing this decoding for myself in a python language. my goal is to merge two friends files but currently i am stuck at the part where you do :
<html>
<body>

int8[] |For each tile ID with multiple options, in sequence, the number of options is recorded.
--
int8[] | For each wall ID with multiple options, in sequence, the number of options is recorded.

</body>
</html>

**Comment by franktreppromo65** [comment]
[MapMerger.zip](https://github.com/TruePikachu/terraria-map-dump/files/9384380/MapMerger.zip)
so far i have this done

**Comment by TruePikachu** [comment]
[I actually had ported it all over to Python+numpy](https://gist.github.com/TruePikachu/96d45978b8311ff3d0563c638f16ae82), though it isn't in a format ready for any sort of release, and I don't quite plan to do more development on it unless I have a good reason to do so. [This is the line which does that particular part of the reading.](https://gist.github.com/TruePikachu/96d45978b8311ff3d0563c638f16ae82#file-tmapdump-py-L41)

    def iter_bits(bytes_obj):
        for byte in bytes_obj:
            for n in (0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80):
                yield byte&n==n
    
    # tile_has_multiple_options is the raw byte array read from that position in the header
    tile_option_count = [
        (f.read(1)[0] if has_mo else 1)
        for i,has_mo in zip(range(n_tiles),iter_bits(tile_has_multiple_options))
    ]

In a nutshell, the array of `int8` that's at that point of the map file is a variable-length array, with length equal to the number of bits set in the prior bit arrays.

**Comment by franktreppromo65** [comment]
that is perfect ! i sure will get a look around and it will debug my code. if you want to callab with me, i want to merge two friends file of a same map so we can see what other have discovered in a world. i will be working on this repo:https://github.com/franktreppromo65/TerrariaMapMerging
